### ISSUE TITLE
Add loss progression information for prompt tuning and fine-tuning

### ISSUE THREAD

**Issue by gkumbhat** [caikit/caikit-nlp#179]
### Description

To be able to provide information about their training progression to enable them to get an idea if training went in right direction or not, we want to give them access to training loss per epoch / step once the training is finished. These logs will be saved alongside the model artifacts in the `model_path` provided to the `save` method.

This needs to be implemented in both prompt tuning and fine-tuning module.

**Comment by gkumbhat** [comment]
Implemented for both fine-tuning and prompt tuning. Closing out this story.
